generator/src/Main.purs

```diff
@@ -0,0 +1,198 @@
+    , { n: "four",      c: "Digit4",     m: [None, Alt, Shift,      AltShift, AltMod,           AltShiftMod] }
+    , { n: "five",      c: "Digit5",     m: [None, Alt, Shift,      AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "six",       c: "Digit6",     m: [None, Alt, Shift,      AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "seven",     c: "Digit7",     m: [None, Alt, Shift,      AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "eight",     c: "Digit8",     m: [None, Alt, Shift,      AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "nine",      c: "Digit9",     m: [None, Alt, Shift,      AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "space",     c: "Space",      m: [None, Alt, Shift,      AltShift,         ShiftMod, AltShiftMod] }
+    , { n: "tab",       c: "Tab",        m: [None,      Shift,                AltMod,           AltShiftMod] }
+    , { n: "enter",     c: "Enter",      m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "home",      c: "Home",       m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "end",       c: "End",        m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "pageDown",  c: "PageDown",   m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "pageUp",    c: "PageUp",     m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "up",        c: "ArrowUp",    m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "down",      c: "ArrowDown",  m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "left",      c: "ArrowLeft",  m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "right",     c: "ArrowRight", m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    , { n: "backspace", c: "Backspace",  m: [None, Alt, Shift, Mod, AltShift, AltMod, ShiftMod, AltShiftMod] }
+    ]
+
+keyShortcuts :: Array ShortcutSource
+keyShortcuts =
+  KeyShortcutSource <$>
+    [ { n: "a", k: "a", m: [KeyNone, KeyMod] }
+    , { n: "b", k: "b", m: [KeyNone, KeyMod] }
+    , { n: "c", k: "c", m: [KeyNone, KeyMod] }
+    , { n: "d", k: "d", m: [KeyNone, KeyMod] }
+    , { n: "e", k: "e", m: [KeyNone, KeyMod] }
+    , { n: "f", k: "f", m: [KeyNone, KeyMod] }
+    , { n: "g", k: "g", m: [KeyNone, KeyMod] }
+    , { n: "h", k: "h", m: [KeyNone, KeyMod] }
+    , { n: "i", k: "i", m: [KeyNone, KeyMod] }
+    , { n: "j", k: "j", m: [KeyNone, KeyMod] }
+    , { n: "k", k: "k", m: [KeyNone, KeyMod] }
+    , { n: "l", k: "l", m: [KeyNone, KeyMod] }
+    , { n: "m", k: "m", m: [KeyNone] }
+    , { n: "n", k: "n", m: [KeyNone, KeyMod] }
+    , { n: "o", k: "o", m: [KeyNone, KeyMod] }
+    , { n: "p", k: "p", m: [KeyNone, KeyMod] }
+    , { n: "q", k: "q", m: [KeyNone] }
```

Comment:
As an example we can see that there is no `KeyMod` in the source for `q` shortcuts as `⌘Q` closes applications on some Apple platforms.